tests/test_layer.py

```diff
@@ -524,5 +529,73 @@ def helper_test_update_test_fixtures_by_file(layer):
             assert type(e) == SherlockUpdateTestFixturesByFileError
 
 
+def helper_test_export_all_test_points(layer):
+    """Tests export_all_test_points API."""
+    try:
+        layer.export_all_test_points(
+            "",
+            "Main Board",
+            "Test Points.csv",
+        )
+        pytest.fail("No exception raised when using an invalid parameter")
+    except SherlockExportAllTestPoints as e:
+        assert e.str_itr()[0] == "Export test points error: Project name is invalid."
+
+    try:
+        layer.export_all_test_points(
+            "Tutorial Project",
+            "",
+            "Test Points.csv",
+        )
+        pytest.fail("No exception raised when using an invalid parameter")
+    except SherlockExportAllTestPoints as e:
+        assert e.str_itr()[0] == "Export test points error: CCA name is invalid."
+
+    try:
+        layer.export_all_test_points(
+            "Tutorial Project",
+            "Main Board",
+            "",
+        )
+        pytest.fail("No exception raised when using an invalid parameter")
+    except SherlockExportAllTestPoints as e:
+        assert e.str_itr()[0] == "Export test points error: File path is required."
+
+    if layer._is_connection_up():
+        if platform.system() == "Windows":
+            temp_dir = os.environ.get("TEMP", "C:\\TEMP")
+        else:
+            temp_dir = os.environ.get("TEMP", "/tmp")
+        test_points_file = os.path.join(temp_dir, "test_points.csv")
+
+        try:
+            result = layer.export_all_test_points(
+                "Tutorial Project",
+                "Main Board",
+                test_points_file,
+            )
+
+            assert os.path.exists(test_points_file)
+            time.sleep(0.5)
```

Comment:
This sleep should be removed. Exporting test points is not multi-threaded so it should return a response only after the file is written.